pkg/gui/context/branches_context.go

```diff
@@ -40,11 +40,12 @@ func NewBranchesContext(c *ContextCommon) *BranchesContext {
 		FilteredListViewModel: viewModel,
 		ListContextTrait: &ListContextTrait{
 			Context: NewSimpleContext(NewBaseContext(NewBaseContextOpts{
-				View:       c.Views().Branches,
-				WindowName: "branches",
-				Key:        LOCAL_BRANCHES_CONTEXT_KEY,
-				Kind:       types.SIDE_CONTEXT,
-				Focusable:  true,
+				View:                       c.Views().Branches,
+				WindowName:                 "branches",
+				Key:                        LOCAL_BRANCHES_CONTEXT_KEY,
+				Kind:                       types.SIDE_CONTEXT,
+				Focusable:                  true,
+				NeedsRerenderOnWidthChange: true,
```

Comment:
I hope I caught all views that need this, please double-check!

Reply:
Looks good to me. The tricky thing is that any view where we want per-item loaders, we'll need to remember to set this value to true.

Reply:
That's not quite true. We already have views with per-item loaders (e.g. tags for when you're deleting a remote tag) that don't need this. It only needs to be set on views that render things differently depending on the available width, it doesn't really have anything to do with item loaders at all.

Reply:
Hmm, that assumes that in any list view where we show a loader the loader will be visible regardless of what else appears in the line. There's nothing stopping a user from creating a tag with a really long name for example. And if the pattern is to show the loader after the name of the list item then that's a problem anywhere. However I'll admit that we typically only see super long names for branches. We'll need to keep an eye on this.

Reply:
True; for tags I'd be happy to ignore the problem for now, because tag names are typically short. (And I already put the loader in front of the tag description, which can be very long, not after it, for exactly this reason.) It might actually look better to put the loader at the end for tags too, and truncate the tag description if necessary, but that's more work and maybe it's not worth it. After all, how often do you delete a remote tag?

Reply:
Rarely. Did we ever test showing the loader at the start for branches? Wonder how that looks. If it looked fine, it would mean we could consistently apply that pattern with little effort

Reply:
No, I didn't try this, but for branches I actually like how the loader is at the end, replacing whatever other status was showing there. You start with `↑4↓9`, then this is hidden by "Pushing...", and then the checkmarks appears from under it. I like that.

Reply:
Also, remember that we are doing the truncation not only so that the `Pushing...` is visible, but also so that the `↑4↓9` is visible. So the effort was already there anyway.

Reply:
> Also, remember that we are doing the truncation not only so that the Pushing... is visible, but also so that the ↑4↓9 is visible. So the effort was already there anyway.

For sure; I very much like the improvement even ignoring the loader side of things.

> You start with ↑4↓9, then this is hidden by "Pushing...", and then the checkmarks appears from under it. I like that.

Good point. I'm happy to roll with that